appveyor.yml

```diff
@@ -12,20 +9,14 @@ branches:
 environment:
   global:
     CYG_ROOT: C:/cygwin
-    CYG_CACHE: C:/cygwin/var/cache/setup
-    CYG_MIRROR: http://mirrors.kernel.org/sourceware/cygwin/
-    OPAM_DIR: C:/opam
-
-cache:
-  - C:\opam
 
 install:
-  - set PATH=%OPAM_DIR%\bin;%PATH%
 #  - appveyor DownloadFile "http://cygwin.com/setup-%CYG_ARCH%.exe" -FileName "%CYG_ROOT%\setup-x86.exe"
-  - '"%CYG_ROOT%\setup-x86.exe" -qnNdO -R "%CYG_ROOT%" -s "%CYG_MIRROR%" -l "%CYG_CACHE%" -P make -P gcc-core -P ocaml -P ocaml-compiler-libs -P m4 -P tar -P patch -P flexdll -P pcre >NUL'
+  - ps: '(new-object net.webclient).DownloadString("https://aantron.github.io/binaries/cygwin/x86/ocaml/4.02/install.ps1") | PowerShell -Command -'
+  - ps: '(new-object net.webclient).DownloadString("https://aantron.github.io/binaries/cygwin/x86/camlp4/4.02/install.ps1") | PowerShell -Command -'
+  - ps: '(new-object net.webclient).DownloadString("https://aantron.github.io/binaries/cygwin/x86/opam/1.2/install.ps1") | PowerShell -Command -'
```

Comment:
Not sure what you mean by self-contained. If you mean including the `install.ps1` files themselves, I chose to do it this way so that the installation procedures can be patched in `binaries` rather than having to change all users. In fact, I have to do that right now due to https://ci.appveyor.com/project/Chris00/oasis-9nf6b/build/1.0.58.